src/Microsoft.OData.Core/Json/BufferingJsonReader.cs

```diff
@@ -81,6 +81,64 @@ internal BufferingJsonReader(IJsonReader innerReader, string inStreamErrorProper
             this.currentBufferedNode = null;
         }
 
+        internal bool TryGetValueFromBuffered(string propertyName, bool removeIfExist, out object value)
+        {
+            bool found = false;
+            BufferedNode currentNode = this.bufferedNodesHead;
+            if (this.bufferedNodesHead != null)
+            {
+                do
+                {
+                    if (currentNode.NodeType == JsonNodeType.StartObject ||
+                        currentNode.NodeType == JsonNodeType.StartArray)
+                    {
+                        break;
+                    }
+
+                    if (currentNode.NodeType == JsonNodeType.Property &&
+                        currentNode.Value.ToString() == propertyName &&
+                        currentNode.Next != currentNode) // must have another node

```

Comment:
Why do check whether there's another node after this?

Reply:
Just to see if I understand, property and value are stored in separate node, so if we find the expected property node, we want to make sure there's a value node after it. Is that correct?

Reply:
That's correct. Property and value are two node types and they are stored in separate node.